sdk/maps/azure-maps-search/src/main/java/com/azure/maps/search/MapsSearchClient.java

```diff
@@ -65,1630 +50,545 @@ public final class MapsSearchClient {
+     * Views.
+     * @param resultType The geopolitical concept to return a boundary for. If not specified, the default is
+     * `countryRegion` result type.
+     * @param resolution Resolution determines the amount of points to send back. If not specified, the default is
+     * medium resolution.
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws ErrorResponseException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return this object is returned from a successful Search Nearby Point of Interest call.
+     * @return `GeoJSON GeocodingFeature` object that describe the boundaries of a geographical area.
      */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public SearchAddressResult searchNearbyPointsOfInterest(SearchNearbyPointsOfInterestOptions options) {
-        return this.asyncClient.searchNearbyPointsOfInterest(options).block();
-    }
 
-    /**
-     * Search Nearby Points of Interest
-     * <!-- src_embed com.azure.maps.search.sync.search_nearby -->
-     * <pre>
-     * System.out.println&#40;&quot;Search Nearby Points of Interest:&quot;&#41;;
-     *
-     * &#47;&#47; options
-     * client.searchNearbyPointsOfInterest&#40;
-     *     new SearchNearbyPointsOfInterestOptions&#40;new GeoPosition&#40;-74.011454, 40.706270&#41;&#41;
-     *         .setCountryFilter&#40;Arrays.asList&#40;&quot;US&quot;&#41;&#41;
-     *         .setTop&#40;10&#41;&#41;;
-     *
-     * &#47;&#47; response
-     * client.searchNearbyPointsOfInterestWithResponse&#40;
-     *     new SearchNearbyPointsOfInterestOptions&#40;new GeoPosition&#40;-74.011454, 40.706270&#41;&#41;
-     *         .setCountryFilter&#40;Arrays.asList&#40;&quot;US&quot;&#41;&#41;
-     *         .setTop&#40;10&#41;,
-     *     null&#41;.getStatusCode&#40;&#41;;
-     * </pre>
-     * <!-- end com.azure.maps.search.sync.search_nearby -->
-     *
-     * @param query A pair of coordinates for query.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws ErrorResponseException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return this object is returned from a successful Search Nearby Point of Interest call.
-     */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public SearchAddressResult searchNearbyPointsOfInterest(GeoPosition query) {
-        return this.asyncClient.searchNearbyPointsOfInterest(new SearchNearbyPointsOfInterestOptions(query)).block();
+    public Boundary getPolygons(GeoPosition coordinates, String view,
+                                      BoundaryResultTypeEnum resultType, ResolutionEnum resolution) {
+        return this.asyncClient.getPolygons(coordinates, view, resultType, resolution).block();
```

Comment:
Since this SDK was last developed there is a new pattern for synchronous clients called Sync Stack. The generated implementation client should have APIs for async and sync, so this synchronous client should be calling into the implementation client's synchronous methods instead of blocking on the handwritten asynchronous client.